src/lib/y2partitioner/clients/main.rb

```diff
@@ -44,9 +45,16 @@ def initialize
       #
       # @param allow_commit [Boolean] whether the changes can be stored on disk
       def run(allow_commit: true)
```

Comment:
Not sure how relevant it is, but before the change this method always returned either nil (if the user quited before attempting to commit) or the output of `StorageManager.commit` (which is a boolean indicating whether the commit was successful).

Now it always returns the output of `inhibitors.uninhibit`.

It would be better to keep the behavior.

Reply:
Summarizing my two previous comments, this should be better in terms of respecting the current behavior and avoiding side effects:

```ruby
def run(allow_commit: true)
  return nil if !run_partitioner?

  begin
    inhibitors = Y2Storage::Inhibitors.new
    inhibitors.inhibit

    if partitioner_dialog.run == :next
      allow_commit ? commit : forbidden_commit_warning
    end
  ensure
    inhibitors.uninhibit
  end
end
```

Or maybe, to look closer to the original (and more explicit about returning nil):

```ruby
def run(allow_commit: true)
  return nil if !run_partitioner?

  begin
    inhibitors = Y2Storage::Inhibitors.new
    inhibitors.inhibit

    return nil if partitioner_dialog.run != :next
    allow_commit ? commit : forbidden_commit_warning
  ensure
    inhibitors.uninhibit
  end
end
```